firmware/src/ledModifier.cpp

```diff
@@ -0,0 +1,93 @@
+#include "ledModifier.hpp"
+
+#include <cmath>
+
+#include "pico/stdlib.h"
+
+static HSV *led_array;
+static long *tick;
+const int nr_of_leds = 5;  // OBS, should be inherrited once linked
+
+// used to silence intellisenses weird hate for M_PI
+static const double pi = 3.14159265359;
+
+void setPointers(HSV *led_array_pointer, long *tick_pointer) {
+	led_array = led_array_pointer;
+	tick = tick_pointer;
+}
+
+void setColour(HSV colour, uint8_t bitmask) {
+	for (int i = 0; i < nr_of_leds; i++) {
+		led_array[i] = colour;
+	}
+}
+
+void changeHue(HSV &led, int amount) {
+	led.hue = ((led.hue + amount) + 255) % 255;	 // looping
+}
+
+void changeSaturation(HSV &led, int amount) {
```

Comment:
You can refactor this to a ternary statement.
`led.saturation = (led.saturation + amount < 0) ? 0 : 255`

https://www.w3schools.com/cpp/cpp_conditions_shorthand.asp